src/System Application/App/AI/src/Azure OpenAI/Chat Completion/AOAIChatMessages.Codeunit.al

```diff
@@ -176,4 +177,103 @@ codeunit 7763 "AOAI Chat Messages"
     begin
         exit(AOAIChatMessagesImpl.PrepareHistory(SystemMessageTokenCount, MessagesTokenCount));
     end;
+
+    /// <summary>
+    /// Appends a Tool to the payload.
+    /// </summary>
+    /// <param name="NewTool">The Tool to be added to the payload.</param>
+    /// <remarks>See more details here: https://go.microsoft.com/fwlink/?linkid=2254538</remarks>
+    [NonDebuggable]
+    procedure AddTool(NewTool: JsonObject)
+    var
+        CallerModuleInfo: ModuleInfo;
+    begin
+        NavApp.GetCallerModuleInfo(CallerModuleInfo);
+        AOAIToolsImpl.AddTool(NewTool, CallerModuleInfo);
+    end;
+
+    /// <summary>
+    /// Modifies a Tool in the list of Tool.
+    /// </summary>
+    /// <param name="Id">Id of the message.</param>
+    /// <param name="NewTool">The new Tool.</param>
+    /// <error>Message id does not exist.</error>
+    [NonDebuggable]
+    procedure ModifyTool(Id: Integer; NewTool: JsonObject)
+    begin
+        AOAIToolsImpl.ModifyTool(Id, NewTool);
+    end;
+
+    /// <summary>
+    /// Deletes a Tool from the list of Tool.
+    /// </summary>
+    /// <param name="Id">Id of the Tool.</param>
+    /// <error>Message id does not exist.</error>
+    [NonDebuggable]
+    procedure DeleteTool(Id: Integer)
+    begin
+        AOAIToolsImpl.DeleteTool(Id);
+    end;
+
+    /// <summary>
+    /// Gets the list of Tools.
+    /// </summary>
+    /// <returns>List of Tools.</returns>
+    [NonDebuggable]
+    procedure GetTools(): List of [JsonObject]
+    begin
+        exit(AOAIToolsImpl.GetTools());
+    end;
+
+    /// <summary>
+    /// Checks if at least one Tools exists in the list.
+    /// </summary>
+    /// <returns>True if Tools exists, false otherwise.</returns>
+    [NonDebuggable]
+    procedure ToolExists(): Boolean
```

Comment:
Rename to `ToolsExists`. Using `ToolExists` sounds like I can check if a specific tool exists.